src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipelineOptions.cs

```diff
@@ -5,131 +5,38 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
-using System.Reflection;
-using Azure.Base.Attributes;
 
 namespace Azure.Base.Http
 {
     public class HttpPipelineOptions
     {
-        ServiceProvider _container;
-
-        HttpPipelineTransport _transport;
-        List<HttpPipelinePolicy> _perCallPolicies;
-        List<HttpPipelinePolicy> _perRetryPolicies;
+        private HttpPipelineTransport _transport = HttpClientTransport.Shared;
 
         public HttpPipelineTransport Transport {
             get => _transport;
-            set {
-                if (value == null) throw new ArgumentNullException(nameof(value));
-                _transport = value;
-            }
+            set => _transport = value ?? throw new ArgumentNullException(nameof(value));
         }
 
-        public HttpPipelinePolicy LoggingPolicy { get; set; }
-
-        public HttpPipelinePolicy RetryPolicy { get; set; }
-
         public bool DisableTelemetry { get; set; } = false;
 
         public string ApplicationId { get; set; }
 
-        public HttpPipelineOptions(HttpPipelineTransport transport)
-            => _transport = transport;
-
-        public HttpPipelineOptions() : this( HttpClientTransport.Shared)
-        { }
-
-        public void AddPerCallPolicy(HttpPipelinePolicy policy)
-        {
-            if (policy == null) throw new ArgumentNullException(nameof(policy));
-
-            if (_perCallPolicies == null) _perCallPolicies = new List<HttpPipelinePolicy>();
-            _perCallPolicies.Add(policy);
-        }
+        public IServiceProvider ServiceProvider { get; set; } = EmptyServiceProvider.Singleton;
 
-        public void AddPerRetryPolicy(HttpPipelinePolicy policy)
-        {
-            if (policy == null) throw new ArgumentNullException(nameof(policy));
+        public IList<HttpPipelinePolicy> BeforePolicies { get; } = new List<HttpPipelinePolicy>();
```

Comment:
The names `BeforePolicies` and `AfterPolicies` don't seem to offer much context.  From this sample, I don't understand what their purpose is nor when I'd add a policy using one over the other.